### PR TITLE
User and roles test

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

File: src/api/app/helpers/webui/webui_helper.rb [hunk 1]
```diff
@@ -261,10 +261,10 @@ def user_and_role(user, role = nil, options = {})
       icon = user_icon(user)
     end
 
-    if !(real_name.empty? || opt[:short])
-      printed_name = "#{real_name} (#{user})"
-    else
+    if real_name.empty? || opt[:short]
```
Comment (Ana06) [inline src/api/app/helpers/webui/webui_helper.rb]: This is not a test, this is changing the code. Why? and why is in the test commit? :thinking:

File: src/api/spec/helpers/webui/webui_helper_spec.rb [hunk 2]
```diff
@@ -320,53 +320,42 @@
 
   describe '#user_and_role' do
     let(:user) { create(:user) }
-    let(:nobody) { create(:user_nobody) }
+    let(:logged_in_user) { create(:user) }
+    let(:anonymous_user) { create(:user_nobody) }
```
Comment (Ana06) [inline src/api/spec/helpers/webui/webui_helper_spec.rb]: why calling `anonymous_user` to the `nobody_user`?

File: src/api/app/helpers/webui/webui_helper.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/app/helpers/webui/webui_helper.rb]: I am not the right one to answere this, since I did not write the code. But if you want I can remove that part from the commit.

File: src/api/spec/helpers/webui/webui_helper_spec.rb (hunk 2 above)
Comment (bgeuken) [inline src/api/spec/helpers/webui/webui_helper_spec.rb]: nobody_user is just the factory name. Though internally we have a user with login name '_nobody_', but to me this shouldn't matter for a test.
We have 2 use cases here, one is logged in users and the other one is anonymous or guest users.

File: src/api/app/helpers/webui/webui_helper.rb (hunk 1 above)
Comment (Ana06) [inline src/api/app/helpers/webui/webui_helper.rb]: I was discussing something related to that with @mschnitzer and he changed that to check something. So I would say that he commited it by accident.

Comment (bgeuken) [inline src/api/app/helpers/webui/webui_helper.rb]: It's just swapping the condition around (removing the !).

Comment (bgeuken) [inline src/api/app/helpers/webui/webui_helper.rb]: @Ana06 Moved it to a separate commit now

**Comment by Ana06** [comment]
Refacto**r** in the commit message
